src/test/test-env-util.c

```diff
@@ -452,4 +452,13 @@ TEST(getenv_steal_erase) {
         assert_se(r > 0);
 }
 
+TEST(strv_env_name_is_valid) {
+        const char *valid_env_names[] = {"HOME", "USER", "SHELL", "PATH", NULL};
+        const char *invalid_env_names[] = {"", "PATH", "home", "user", "SHELL", NULL};
+        const char *repeated_env_names[] = {"HOME", "USER", "SHELL", "USER", NULL};
+        assert_se(strv_env_name_is_valid((char **) valid_env_names));
+        assert_se(!strv_env_name_is_valid((char **) invalid_env_names));
+        assert_se(!strv_env_name_is_valid((char **) repeated_env_names));
```

Comment:
This should use `STRV_MAKE()` instead of constructing the strvs manually.

Reply:
-> #27108